docs/form.rst

```diff
@@ -0,0 +1,271 @@
+
+The first line creates a "Form" object that is assigned into variable `$f`. Next
+line defines a new field that is placed inside a form.
+
+Once form is defined, it needs to be placed somewhere in a Render Tree.
+
+.. php:method:: addField(name)
+
+    Create a new field on a form. If form is associated with a model, which have
+    a field with a matching name, then field meta-information will be loaded from
+    the model.
+
+If $form is not yet associated with a model (like above) then an empty model will
+be created. The arguments to addField are compatible with Model::addField()::
+
+    $form->addField('is_accept_terms', ['type'=>'boolean']);
+
+Additionally, any fields that are added into Form directly are marked with the
+flag "never_persist".
+
+.. php:method:: setModel($model, [$fields])
+
+    Associate field with existing model object and import all editable fields
+    in the order in which they were defined inside model's init() method.
+
+    You can specify which fields to import and their order by simply listing
+    field names through second argument.
+
+    Specifying "false" or empty array as a second argument will import no fields.
+
+.. php:attr:: model
+
+    Model that is currently associated with a Form.
+
+For the next demo, lets actually define a model::
+
+    class Person extends \atk4\data\Model
+    {
+        public $table = 'person';
+
+        public function init()
+        {
+            parent::init();
+            $this->addField('name');
+            $this->addField('surname');
+            $this->addField('gender', ['enum' => ['M', 'F']]);
+        }
+    }
+
+We can now populate form fields based around the fields as they are defined inside
+a model. I will also add one extra checkbox where user can accept terms and conditions::
+
+    $form = $layout->add('Form'); // using short version
+
+    $form->setModel(new Person($db));
+
+    $form->addField(
+        'terms', 
+        ['type'=>'boolean', 'ui'=>['caption'=>'Accept Terms and Conditions']]
+    );
+
+Form Submit Handling
+--------------------
+
+.. php:method:: onSubmit($callback)
+
+    Specify a PHP call-back that will be executed on successful form submission. 
+
+.. php:method:: error($field, $message)
+
+    Create and return :php:class:`jsChain` action that will indicate error on a field.
+
+.. php:method:: success($title, [$sub_title])
+
+    Create and return :php:class:`jsChain` action, that will replace form with a success message.
+
+.. php:attr:: successTemplate
+
+    Name of the template which will be used to render success message.
+
+To continue with my example, I'd like to add new Person record into the database
+but only if they have also accepted terms and conditions. I can define onSubmit handler
+that would perform the check, display error or success message::
+
+    $form->onSubmit(function($f) {
+        if (!$f->model['terms']) {
+            return $form->error('terms', 'You must accept terms and conditions');
```

Comment:
$f, because it's inside a callback as parameter. I'm not sure which is better, to let users be ignorant or hit them with a stick.